test/plugin/base_test.rb

```diff
@@ -1045,6 +1045,188 @@ def test_log_entry_trace_field
+  def test_docker_container_application_logs
+    setup_gce_metadata_stubs
+    setup_metadata_agent_stubs
+    setup_logging_stubs do
+      # Metadata Agent is not enabled. Will call Docker Remote API for
+      # container info.
+      d = create_driver(ENABLE_METADATA_AGENT_CONFIG)
+      d.emit(docker_container_application_log_entry(log_entry(0)))
+      d.run
+    end
+    verify_log_entries(1, DOCKER_CONTAINER_PARAMS_WITH_NO_STREAM)
+    assert_requested_metadata_agent_stub(IMPLICIT_LOCAL_RESOURCE_ID)
+    assert_requested_metadata_agent_stub(
+      "containerName.#{DOCKER_CONTAINER_NAME}")
+  end
+
+  # Test that the 'time' field from the json record is extracted and set to
+  # entry.timestamp for Docker container logs.
+  def test_time_field_extraction_for_docker_container_logs
+    setup_gce_metadata_stubs
+    setup_metadata_agent_stubs
+    setup_logging_stubs do
+      d = create_driver(ENABLE_METADATA_AGENT_CONFIG)
+      d.emit(docker_container_application_log_entry(log_entry(0)))
+      d.run
+    end
+    verify_log_entries(1, DOCKER_CONTAINER_PARAMS_WITH_NO_STREAM) do |entry|
+      # Timestamp in 'time' field from log entry should be set properly.
+      assert_equal DOCKER_CONTAINER_SECONDS_EPOCH,
+                   entry['timestamp']['seconds'], entry
+      assert_equal DOCKER_CONTAINER_NANOS, entry['timestamp']['nanos'],
+                   entry
+    end
+    assert_requested_metadata_agent_stub(IMPLICIT_LOCAL_RESOURCE_ID)
+    assert_requested_metadata_agent_stub(
+      "containerName.#{DOCKER_CONTAINER_NAME}")
+  end
+
+  # Test that the 'source' field is properly extracted from the record json and
+  # set as a common label 'stream'. Also entry.severity is set accordingly for
+  # Docker container logs.
+  def test_source_and_severity_for_docker_container_logs
+    setup_gce_metadata_stubs
+    setup_metadata_agent_stubs
+    {
+      docker_container_stdout_stderr_log_entry(
+        log_entry(0), DOCKER_CONTAINER_STREAM_STDOUT) =>
+        [DOCKER_CONTAINER_PARAMS, 'INFO'],
+      docker_container_stdout_stderr_log_entry(
+        log_entry(0), DOCKER_CONTAINER_STREAM_STDERR) =>
+        [DOCKER_CONTAINER_PARAMS_WITH_STREAM_STDERR, 'ERROR'],
+      docker_container_application_log_entry(log_entry(0)) =>
+        [DOCKER_CONTAINER_PARAMS_WITH_NO_STREAM, 'DEFAULT'],
+      docker_container_application_log_entry(log_entry(0)) \
+        .merge('severity' => 'warning') =>
+        [DOCKER_CONTAINER_PARAMS_WITH_NO_STREAM, 'WARNING']
+    }.each do |log_entry, (expected_params, severity)|
+      @logs_sent = []
+      setup_logging_stubs do
+        d = create_driver(DOCKER_CONTAINER_CONFIG)
+        d.emit(log_entry)
+        d.run
+      end
+      verify_log_entries(1, expected_params) do |entry|
+        assert_equal_with_default entry['severity'], severity, 'DEFAULT', entry
+      end
+    end
+  end
+
+  # Test logs from applications running in GKE. These logs has the label
```

Comment:
Nit: `These logs have the label`... But I wonder why you chose to call it out for GKE, but not for Docker...

Reply:
Changed `has` to `have` and called it out for Docker container as well.